raptorWeb/authprofiles/views.py

```diff
@@ -0,0 +1,347 @@
+            return render(request, template_name, context=dictionary)
+        else:
+            return HttpResponseRedirect('../')
+
+    def post(self, request):
+
+        login_form = UserLoginForm(request.POST)
+        dictionary = player_poller.currentPlayers_DB
+        dictionary["login_form"] = self.login_form
+
+        if login_form.is_valid():
+            username = login_form.cleaned_data["username"]
+            password = login_form.cleaned_data["password"]
+            user = authenticate(username=username, password=password)
+            if user:
+                LOGGER.info("User logged in!")
+                login(request, user)
+                return HttpResponseRedirect(self.request.META.get('HTTP_REFERER'))
+            else:
+                return HttpResponse("Account does not exist")
+        else:
+            messages.error(request, login_form.errors.as_text().replace('* __all__', ''))
+            return HttpResponseRedirect(self.request.META.get('HTTP_REFERER'))
+
+class UserLogin_OAuth(TemplateView):
+
+    def get(self, request):
+        return redirect(settings.DISCORD_AUTH_URL)
+
+class UserLogin_OAuth_Success(TemplateView):
+
+    def get(self, request):
+
+        try:
+            discord_code = request.GET.get('code')
+            user_info = discordAuth.exchange_code(discord_code)
+            discord_user = authenticate(request, user=user_info)
+            try:
+                login(request, discord_user, backend='authprofiles.auth.DiscordAuthBackend')
+            except AttributeError:
+                login(request, list(discord_user).pop(), backend='authprofiles.auth.DiscordAuthBackend')
+            return redirect('../../')
+        except KeyError:
+            return HttpResponseRedirect("../login")
+
+@login_required
+def user_logout(request):
+    """
+    Log out the signed in user
+    """
+    logout(request)
+    LOGGER.info("User logged out!")
+    return HttpResponseRedirect(request.META.get('HTTP_REFERER'))
+
+class All_User_Profile(TemplateView):
+    """
+    Displays all User Profiles
+    """
+    template_name = join(settings.AUTH_TEMPLATE_DIR, 'all_profiles.html')
+
+    def get(self, request):
+        instance_dict = player_poller.currentPlayers_DB
+        instance_dict = viewContext.update_context(context = instance_dict)
+
+        return render(request, self.template_name, context=instance_dict)
+
+class User_Profile(TemplateView):
+    """
+    Displays a User's Profile and it's information
+    """
+    template_name = join(settings.AUTH_TEMPLATE_DIR, 'profile.html')
+
+    def get(self, request, profile_name):
+        instance_dict = player_poller.currentPlayers_DB
+        instance_dict = viewContext.update_context(context = instance_dict)
+        try:
+            user_base = User.objects.get(username=profile_name)
+            user_extra = UserProfileInfo.objects.get(user=user_base)
+            try:
+                if settings.DEBUG:
```

Comment:
There is undoubtedly a cleaner way to write lines 163-227 while still accounting for dev/production environments. There is far too much code duplication here.

Perhaps the actual User/UserInfo and DiscordUserInfo objects themselves can be put into context and read by Django directly, instead of parsing their information into dictionaries. URLs that change depending on environment need to be set in `settings` module rather than directly changing them in code based on the status of `settings.DEBUG`